pkg/setting/setting_oauth.go

```diff
@@ -17,6 +17,19 @@ type OAuthInfo struct {
 	TlsClientKey           string
 	TlsClientCa            string
 	TlsSkipVerify          bool
+	GroupMappingsFile      string
+	GroupMappings          []OAuthGroupMapping
+}
+
+type OAuthGroupMapping struct {
+	RoleAttributePath string `toml:"role_attribute_path"`
+	Role              string
+	OrgId             int   `toml:"org_id"`
+	IsGrafanaAdmin    *bool `toml:"grafana_admin"`
```

Comment:
Should need pointer since false is the default if not provided.

```suggestion
	IsGrafanaAdmin    bool `toml:"grafana_admin"`
```

Reply:
I ran into a lot of problems when testing this when not using a pointer.  I could only get the Grafana Admin assignment to work when I used a pointer.  I stumbled across that solution when looking at how the LDAP code assigned Grafana Admin.  I think it might have to do with what `extUser.IsGrafanaAdmin` expects `*bool` for `IsGrafanaAdmin`:

```go
type ExternalUserInfo struct {
	OAuthToken     *oauth2.Token
	AuthModule     string
	AuthId         string
	UserId         int64
	Email          string
	Login          string
	Name           string
	Groups         []string
	OrgRoles       map[int64]RoleType
	IsGrafanaAdmin *bool // This is a pointer to know if we should sync this or not (nil = ignore sync)
	IsDisabled     bool
}
```

Reply:
Would is possibly work to make `IsGrafanaAdmin` from TOML not a pointer and then assign by doing this: `extUser.IsGrafanaAdmin = &groupMapping.IsGrafanaAdmin`, or should I just keep things a pointer?

Reply:
Sorry, you're right. My bad, let's keep the pointer